### PR TITLE
wrap the message part of tracy and provide a color "framework" based on `printstyle`

### PR DESCRIPTION
**Pull request by KristofferC**
Here are some screenshots from the tests:

Screenshot 2023-05-16 at 21 18 20

### CODE REVIEW

**Comment by KristofferC** [comment]
> The signed vs. unsigned integer distinction is kind of surprising. What do you think about dropping the printstyled(; color::Int) behavior, so that we just have the unsigned behavior for Integer?

I was thinking about that too. The reason for adding that was to keep things consistent with `printstyled`. So if you did

```
msg = "Here is a msg"
color = 5
tracymsg(msg color)
printstyled(msg; color)
```

You would get the same in your terminal as in tracy.

**Comment by KristofferC** [comment]
The usage here is identical to that in https://github.com/KristofferC/Crayons.jl (which is perhaps not surprising since I made that one...). But that might be somewhat of an advantage to have `printstyled´, `Crayones` and this package all somewhat agree.

In the vast majority of the cases you won't really do a computation to get your color so even while the unsigned branch feels a bit unsettling, I am not sure how much of an issue it will be in practice.

**Comment by KristofferC** [comment]
Error during backtrace collection on mac?

```
[2999] signal (11.1): Segmentation fault: 11
in expression starting at /Users/runner/work/Tracy.jl/Tracy.jl/test/run_zones.jl:34
__thread_stack_pcs at /usr/lib/system/libsystem_c.dylib (unknown line)
_thread_stack_pcs at /usr/lib/system/libsystem_c.dylib (unknown line)
backtrace at /usr/lib/system/libsystem_c.dylib (unknown line)
___tracy_emit_messageC at /Users/runner/.julia/artifacts/d67f2bdc06cf897f67dfa47bab1dee1740780697/lib/libTracyClient.dylib (unknown line)
```

**Comment by topolarity** [comment]
Another option is to make `color` behave like printstyled and add `raw_color` that supports RGB triplets/unsigned hex values

**Comment by KristofferC** [comment]
> My request would be to either split printstyled colors out to a terminal_color argument or TerminalColor wrapper type, or break terminal compatibility and leave the mapping for someone else to implement

Okay, maybe having the terminal compatibility stuff is just pointless?

Would be a lot simpler if all we provided was `UInt` and `(r, g, b)`.

**Comment by KristofferC** [comment]
Last commits removes the `printstyled` equivalent number.

File: src/msg.jl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  `:light_black`, `:light_blue`, `:light_green`, `:light_cyan`, `:light_red`, `:light_magenta`, `:light_yellow`, `:light_white`.
+- A tuple of three integers: The RGB value `(R, G, B)` where each value is in the range 0..255.
+
+The `callstack` argument determines whether a callstack is collected.
+"""
+function tracymsg(msg::AbstractString; color::Union{Integer,Symbol,NTuple{3,Integer},Nothing}=nothing, callstack::Bool=false)
+    if color === nothing
+        @ccall libtracy.___tracy_emit_message(msg::Cstring, length(msg)::Csize_t, callstack::Cint)::Cvoid
+    else
+        @ccall libtracy.___tracy_emit_messageC(msg::Cstring, length(msg)::Csize_t, _tracycolor(color)::UInt32, callstack::Cint)::Cvoid
```
Comment (topolarity) [inline src/msg.jl:18]: I think the `callstack` here is actually a max stack depth to record

**Review by topolarity — APPROVED** [review]
A very nice enhancement 

We might do a "naming pass" later and revise `@tracepoint` and possibly `tracymsg` etc., but for now I'm happy to be clocking in features 😄

**Comment by KristofferC** [comment]
> We might do a "naming pass" later

Agree with that. As you say probably easier to pick good names once all pieces are in place.

**Pull request merged**